Lib/importlib/_bootstrap_external.py

```diff
@@ -1475,7 +1475,10 @@ def _get_spec(cls, fullname, path, target=None):
         #  the list of paths that will become its __path__
         namespace_path = []
         for entry in path:
-            if not isinstance(entry, str):
+            try:
+                # bytes, str and path-like objects will pass fspath
```

Comment:
you need to add a test for bytes on `sys.path`, and test it with the other import systems. Note these tests fail with a BytesWarning